install.sh

```diff
@@ -1,47 +1,49 @@
 #!/bin/bash -e
 
 OLM_VERSION=0.17.0
-OPERATOR_INDEX_IMAGE=${OPERATOR_INDEX_IMAGE:-registry.redhat.io/redhat/community-operator-index:latest}
-OPERATOR_REGISTRY_REF=$(echo $OPERATOR_INDEX_IMAGE | cut -f 1 -d '/')
 OPERATOR_CHANNEL=${OPERATOR_CHANNEL:-beta}
 DOCKER_CFG=${DOCKER_CFG:-$HOME/.docker/config.json}
 CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-docker}
 
-#The SBO can also be installed in a vanilla kubernetes cluster. A prerequisite for this would be to add credentials for the registry.redhat.io in the cluster. Steps to be followed to achieve the same are listed below:
-#1) Follow [Red Hat Container Registry Authentication Steps](https://access.redhat.com/RegistryAuthentication)
-#2) Verify that your credentials are correct using docker login -u <your_username> -p <your_passwd> registry.redhat.io
-
-if [ -z "$SKIP_REGISTRY_LOGIN" ]; then
-  if [ -z "$OPERATOR_REGISTRY_USERNAME" ]; then
-    ${CONTAINER_RUNTIME} login $OPERATOR_REGISTRY_REF
-  else
-    ${CONTAINER_RUNTIME} login -u "$OPERATOR_REGISTRY_USERNAME" --pasword-stdin $OPERATOR_REGISTRY_REF <<<$OPERATOR_REGISTRY_PASSWORD
-  fi
+if [ "$(kubectl get crd clusterserviceversions.operators.coreos.com &>1 || echo false)" == "false" ]; then
+  curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v${OLM_VERSION}/install.sh | bash -s v${OLM_VERSION}
 fi
 
-#3) Start a kubernetes cluster. Any Kubernetes cluster can be used as well. The script assumes that a Kubernetes is up and running and the user has logged into it.
+CATSRC_NAMESPACE="${CATSRC_NAMESPACE:-$(kubectl get catalogsources.operators.coreos.com --all-namespaces -o jsonpath='{.items[0].metadata.namespace}' --ignore-not-found=true)}"
 
-#4) Enable [OLM](https://github.com/operator-framework/operator-lifecycle-manager) in the cluster by running the following command
-curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v${OLM_VERSION}/install.sh | bash -s v${OLM_VERSION}
-#On minikube you can alternatively install OLM by running `minikube addons enable olm`
+OPNS="$(kubectl get namespaces -o name | grep -E '.*/(.+-)?operators$' || test $? = 1)"
+OPERATOR_NAMESPACE=${OPNS#"namespace/"}
 
-#5) Create a new image pull secret out of your local .docker/config.json file
-if [ -r "$DOCKER_CFG" ]; then
-  kubectl create secret generic sbo-operators-secrets -n olm --from-file=.dockerconfigjson=$DOCKER_CFG --type=kubernetes.io/dockerconfigjson
+if [ -n "$OPERATOR_INDEX_IMAGE" ]; then
+  if [ -r "$DOCKER_CFG" ]; then
+    kubectl create secret generic sbo-operators-secrets -n $CATSRC_NAMESPACE --from-file=.dockerconfigjson=$DOCKER_CFG --type=kubernetes.io/dockerconfigjson --dry-run=client -o json | kubectl apply -f -
```

Comment:
why do we need to use `--dry-run` option now and pass the result to another `kubectl apply`?